src/Microsoft.DotNet.Build.Tasks/GenerateCurrentVersion.cs

```diff
@@ -70,6 +93,17 @@ public override bool Execute()
             return true;
         }
 
+        public string GetCurrentRevisionForBuildName(string buildName)
+        {
+            Regex regex = new Regex(@"^\d*\.(\d{2})$");
+            Match match = regex.Match(buildName);
+            if (match.Success && match.Groups.Count > 1)
+            {
+                return match.Groups[1].Value;
+            }
+            return "00";
```

Comment:
`\.\d+$` that should be enough for the regex. Thinking a little more if this doesn't doesn't exist lets default to just "0" that way we can actually use it as an indicator for something different for local builds.